src/Maybe.gren

```diff
@@ -203,6 +254,28 @@ map5 func ma mb mc md me =
                                             Just (func a b c d e)
 
 
+{-| Returns `Nothing` if the contained value doesn't pass the given
+test.
+
+    filter isOdd (Just 5) == Just 5
+
+    filter isOdd (Just 2) == Nothing
+
+-}
+filter : (a -> Bool) -> Maybe a -> Maybe a
+filter test maybe =
```

Comment:
Would `keep` of `keepIf` be potential better names? I think with keepIf, it's clearer (especially to beginners) what the function will do if the predicate returns false (which may require some learning for filter).

Reply:
I personally think `keepIf` is a better name, but I also value consistency so I like similar functions to be similarly named. Since it's called `filter` in other modules, it should be `filter` here as well.

Fun fact: Elm did briefly consider to rename `filter` to `keep`: https://github.com/elm/core/commit/77e5c6711c2f956e14ede5b08a3ae7dbb0464f6c#diff-0cffa339bd6235b9902e9934de9ddd8f

I guess there's an argument to be made that _every other language_ calls this operation `filter`, so having it named something else in Gren could be confusing?

Reply:
Yes, I mentioned it because I know Evan thought about it (and regretted not naming it that way), although I didn't know there was a commit for it.

> I guess there's an argument to be made that every other language calls this operation filter, so having it named something else in Gren could be confusing?

Right. Although Elm renamed `bind` to `andThen`. And for `List` it's `concatMap`. I don't know the right answer either. And if it were to be renamed, I'd suggest at least dropping the word "filter" in the documentation.

And maybe even make it so that in the documentation website, if people like for `filter`, then `keep` shows up?
```elm
{-| bla bla bla explanation

Also known as `filter` in other languages.

@doc-aliases filter, keepIf
-}
keep : (a -> Bool) -> Maybe a -> Maybe a
keep = ...
```
but that's obviously more involved.